classes/assessment/lesson.php

```diff
@@ -0,0 +1,62 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace local_sitsgradepush\assessment;
+
+/**
+ * Class for lesson assessment.
+ *
+ * @package    local_sitsgradepush
+ * @copyright  2024 onwards University College London {@link https://www.ucl.ac.uk/}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @author     David Watson <david-watson@ucl.ac.uk>
+ */
+class lesson extends activity {
+
+    /**
+     * Get all participants.
+     *
+     * @return array
+     */
+    public function get_all_participants(): array {
+        if ($this->sourceinstance->practice) {
```

Comment:
Is it possible to prevent a ‘Practice Lesson’ from being displayed on the ‘Select Source’ page?

Reply:
I opened PR #67 to address this since the change involves multiple classes so wouldn't work in this branch